Zend/zend_language_scanner.l

```diff
@@ -1956,6 +1957,54 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	}
 }
 
+<ST_IN_SCRIPTING>{ONUM} {
+	/* The +/- 2 skips "0o" */
+	size_t len = yyleng - 2;
+	char *end, *octal = yytext + 2;
+	zend_bool contains_underscores = (memchr(octal, '_', len) != NULL);
+
+	/* Skip any leading 0s */
+	while (len > 0 && (*octal == '0' || *octal == '_')) {
+		++octal;
+		--len;
+	}
+
+	if (contains_underscores) {
+		octal = estrndup(octal, len);
+		strip_underscores(octal, &len);
+	}
+
+	/* Computation of length of string
+	 * MAX_LENGTH_OF_LONG = 4 on 32bits and 8 on 64bits
+	 * 17777777777 32bit octal max signed integer (11 chars): 4 * 2 + (4%3 *2) | equal to 10
+	 * 1000000000000000000000 64bit max signed integer (22 chars): 8 * 2 + (8%3 *2) | equal to 20
+	 * We let the 64bit case be degenerate in only accepting max value - 1 (i.e. 777777777777777777777)
```

Comment:
I don't get this comment. Why is the 64bit case special?

Personally I'd prefer to just call ZEND_STRTOL always here and check for ERANGE afterwards, rather than doing manual length checks.

Reply:
Ah that's a bit of a leftover, I was looking at how the Hex and Binary literals were implemented and they did a manual check, and I thought I had found out a formula but miscounted the number of zeros in a 64bit Octal integer literal.

The check for ERANGE seems reasonable as this is partly how the decimal/general case handles it, will look into that.